.github/workflows/operator_ci.yaml

```diff
@@ -0,0 +1,181 @@
+name: Operator Build
+on:
+  push:
+    branches:
+      - develop
+      - master
+  pull_request:
+    branches:
+      - develop
+      - master
+jobs:
+  prepare_env:
+   name: "Prepare env"
+   runs-on: ubuntu-latest
+   steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9.7'
+      - uses: FranzDiebold/github-env-vars-action@v2
+      - name: Setup dependencies
+        id: setup_dependencies
+        run: |
+          build/ci/github_actions/setup_ci_dependencies.sh
+        env:
+          build_number: "${{ github.run_number }}"
+      - uses: actions/cache@v2
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ hashFiles('dev-requirements.txt') }}
+      - name: Save dependencies files
+        uses: actions/upload-artifact@v2
+        with:
+          name: ci_dependencies
+          path: |
+            /home/runner/.bash_profile
+          retention-days: 1
+   outputs:
+     branch_image_tag: "${{ steps.setup_dependencies.outputs.branch_image_tag }}"
+     operator_specific_tag_for_test: "${{ steps.setup_dependencies.outputs.operator_specific_tag_for_test }}"
+
+  unit_testing:
+    runs-on: ubuntu-latest
+    name: "Unit testing"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Run unit tests
+        run: |
+          mkdir -p build/_output/reports && chmod 777 build/_output/reports
+          ./build/ci/run_unittests.sh
+
+  run_olm_validation:
+    runs-on: ubuntu-latest
+    name: "Run OLM validation"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Run OLM validation
+        run: |
+          ./build/ci/run_olm_validation.sh
+          
+  build_and_push_images:
+    name: "Build and push images"
+    runs-on: ubuntu-latest
+    needs:
+      - unit_testing
+      - run_olm_validation
+      - prepare_env
+    strategy:
+      matrix:
+        image_type: ['operator']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1.3.0
+      - uses: FranzDiebold/github-env-vars-action@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: '${{ secrets.CSIBLOCK_DOCKER_REGISTRY_USERNAME }}'
```

Comment:
I would avoid using the current registry/user name (csiblock) in our variables, since it can always change in the future